internal/resources/metal/gateway/models.go

```diff
@@ -8,54 +8,8 @@ import (
 	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
 )
 
-type ResourceModel struct {
-	ID                    types.String   `tfsdk:"id"`
-	ProjectID             types.String   `tfsdk:"project_id"`
-	VlanID                types.String   `tfsdk:"vlan_id"`
-	VrfID                 types.String   `tfsdk:"vrf_id"`
-	IPReservationID       types.String   `tfsdk:"ip_reservation_id"`
-	PrivateIPv4SubnetSize types.Int64    `tfsdk:"private_ipv4_subnet_size"`
-	State                 types.String   `tfsdk:"state"`
-	Timeouts              timeouts.Value `tfsdk:"timeouts"`
-}
-
-func (m *ResourceModel) parse(gw *metalv1.FindMetalGatewayById200Response) diag.Diagnostics {
-	// Convert Metal Gateway data to the Terraform state
-	if gw.MetalGateway != nil {
-		m.ID = types.StringValue(gw.MetalGateway.GetId())
-		m.ProjectID = types.StringValue(gw.MetalGateway.Project.GetId())
-		m.VlanID = types.StringValue(gw.MetalGateway.VirtualNetwork.GetId())
-		m.VrfID = types.StringNull()
-
-		if gw.MetalGateway.IpReservation != nil {
-			m.IPReservationID = types.StringValue(gw.MetalGateway.IpReservation.GetId())
-		} else {
-			m.IPReservationID = types.StringNull()
-		}
-
-		m.PrivateIPv4SubnetSize = calculateSubnetSize(gw.MetalGateway.IpReservation)
-		m.State = types.StringValue(string(gw.MetalGateway.GetState()))
-	} else {
-		m.ID = types.StringValue(gw.VrfMetalGateway.GetId())
-		m.ProjectID = types.StringValue(gw.VrfMetalGateway.Project.GetId())
-		m.VlanID = types.StringValue(gw.VrfMetalGateway.VirtualNetwork.GetId())
-		m.VrfID = types.StringValue(gw.VrfMetalGateway.Vrf.GetId())
-
-		if gw.VrfMetalGateway.IpReservation != nil {
-			m.IPReservationID = types.StringValue(gw.VrfMetalGateway.IpReservation.GetId())
-		} else {
-			m.IPReservationID = types.StringNull()
-		}
-
-		m.PrivateIPv4SubnetSize = calculateSubnetSize(gw.VrfMetalGateway.IpReservation)
-		m.State = types.StringValue(string(gw.VrfMetalGateway.GetState()))
-	}
-	return nil
-}
-
-type DataSourceModel struct {
+type BaseModel struct {
```

Comment:
Is there any reason to export BaseModel? Why not baseModel?

Reply:
Per https://github.com/equinix/terraform-provider-equinix/pull/842#discussion_r1947025206 there doesn't seem to be a reason to export any of the models

Reply:
Lower-cased all of the models affected by this PR.  The AWS provider does the same for its framework models.